test/sensitivities/functional/reducedim.jl

```diff
@@ -3,29 +3,41 @@
         # mapreducedim on a single-dimensional array should be consistent with mapreduce.
         x = Leaf(Tape(), [1.0, 2.0, 3.0, 4.0, 5.0])
         s = 5.0 * mapreduce(abs2, +, x, dims=1)
-        @test ∇(s)[x] ≈ 5.0 * [2.0, 4.0, 6.0, 8.0, 10.0]
+        @test ∇(s, oneslike(s.val))[x] ≈ 5.0 * [2.0, 4.0, 6.0, 8.0, 10.0]
 
         # mapreduce on a two-dimensional array when reduced over a single dimension
         # should give different results to mapreduce over the same array.
         x2_ = reshape([1.0, 2.0, 3.0, 4.0,], (2, 2))
         x2 = Leaf(Tape(), x2_)
         s = mapreduce(abs2, +, x2, dims=1)
-        @test ∇(s, one(s.val))[x2] ≈ 2.0 * x2_
+        @test ∇(s, oneslike(s.val))[x2] ≈ 2.0 * x2_
 
         # mapreducedim under `exp` should trigger the first conditional in the ∇ impl.
         x3_ = randn(rng, 5, 4)
         x3 = Leaf(Tape(), x3_)
         s = mapreduce(exp, +, x3, dims=1)
-        @test ∇(s, one(s.val))[x3] == exp.(x3_)
+        @test ∇(s, oneslike(s.val))[x3] == exp.(x3_)
 
         # mapreducedim under an anonymous-function should trigger fmad.
         x4_ = randn(rng, 5, 4)
         x4 = Leaf(Tape(), x4_)
         s = mapreduce(x->x*x, +, x4, dims=2)
-        @test ∇(s, one(s.val))[x4] == 2x4_
+        @test ∇(s, oneslike(s.val))[x4] == 2x4_
 
         # Check that `sum` works correctly with `Node`s.
         x_sum = Leaf(Tape(), randn(rng, 5, 4, 3))
         @test sum(x_sum, dims=[2, 3]).val == mapreduce(identity, +, x_sum, dims=[2, 3]).val
+
+        # Ensure that the underlying value is correct in the presence of keyword arguments
```

Comment:
Add a test using `mean` too, just so we have a couple things covered and we're not just depending on the specific behaviour of `sum`.

Reply:
Done!